osquery/tables/system/darwin/xprotect.cpp

```diff
@@ -26,11 +26,32 @@ namespace fs = boost::filesystem;
 namespace osquery {
 namespace tables {
 
-/// Path to XProtect.meta.plist and XProtect.plist
-const std::string kXProtectPath =
+/// Path to XProtect.meta.plist and XProtect.plist, before macOS 11
+const std::string kXProtectPathPreBigSur =
     "/System/Library/CoreServices/"
     "CoreTypes.bundle/Contents/Resources/";
 
+/// the path to the XProtect.meta.plist and XProtect.plist changed in macOS 11
+const std::string kXProtectPathPostBigSur =
+    "/Library/Apple/System/Library/"
+    "CoreServices/XProtect.bundle/Contents/Resources";
+
+Status getExpectedXProtectDir(std::string& path) {
+  auto qd = SQL::selectAllFrom("os_version");
+  if (qd.size() != 1) {
+    return Status(-1, "Couldn't determine macOS version");
```

Comment:
Prefer the newer `return Status::failure("error message here")` style, the older one has been deprecated

Reply:
good to know, thanks!